Edit profile form

julesfll opened the pull request:
# What I did
Still working on it
## Issues addressed

# Screenshots

j-alicia-long commented:
@julesfll Make sure you're running ./precommit to check tests & pylint before you push changes so your builds won't fail on Travis :)

julesfll commented:
# Status
Done

# What I did
Remade profile page and added an edit form:
- Removed student id and school
- Replaced with pre-populated form
- Form includes first name, last name, and year
- Will add major later

## Issues addressed
Lack of edit functionality in form

(editing name and year currently has no effect on other features, it is just for display)

## Screenshots
![image](https://user-images.githubusercontent.com/45245651/97786555-c03f3b80-1b82-11eb-8960-4d56c39a1f85.png)

j-alicia-long commented on tcf_website/views/profile.py:
```diff
@@ -2,12 +2,50 @@
+        model = User
+        fields = ['first_name', 'last_name', 'graduation_year']
+
+        # Add the form-control class to make the form work with Bootstrap
```
awesome

j-alicia-long commented on tcf_website/views/profile.py:
```diff
@@ -2,12 +2,50 @@
+
+        if form.is_valid():
+            form.save()
+            messages.success(request, 'Your profile was updated succesfully!')
```
oof fix that typo there

julesfll replied:
```suggestion
            messages.success(request, 'Your profile was updated successfully!')
```

j-alicia-long commented on tcf_core/settings.py:
```diff
@@ -12,6 +12,7 @@
 """
 
 import os
+from django.contrib.messages import constants as messages
```
ay this is pretty cool stuff

j-alicia-long commented on tcf_website/templates/base/messages.html:
```diff
@@ -7,7 +7,7 @@
     <div class="content">
         {% for message in messages %}
-            <div class="alert alert-{{message.tags}} alert-dismissible fade show">
+            <div class="alert {{message.tags}} alert-dismissible fade show">
```
nice modularity

so we can use this for alerts all over the site then yay

j-alicia-long reviewed (commented):
Looks good! Just a quick typo on "successfully" I noticed (might've been from before)

Also I mentioned before that having no other images on the site makes this one look a little out of place, so maybe we'll have to use that collage somewhere hmm

alex-shen1 commented:
Any idea why "Files changed" has several random "empty" files with 0 lines changed listed? Curious about that.

julesfll commented:
> Any idea why "Files changed" has several random "empty" files with 0 lines changed listed? Curious about that.

Yeah I have no idea, I thought it could be a line ending issue but it doesn't seem to be

j-alicia-long reviewed (approved):
I think it's good! Once this gets merged we can modify the form on the collect majors PR
